src/dbnode/storage/index/fields_terms_iterator.go

```diff
@@ -94,65 +94,49 @@ func newFieldsAndTermsIterator(
 	reader segment.Reader,
 	opts fieldsAndTermsIteratorOpts,
 ) (fieldsAndTermsIterator, error) {
-	iter := &fieldsAndTermsIter{}
-	err := iter.Reset(ctx, reader, opts)
-	if err != nil {
-		return nil, err
-	}
-	return iter, nil
-}
-
-func (fti *fieldsAndTermsIter) SearchDuration() time.Duration {
-	return fti.searchDuration
-}
-
-func (fti *fieldsAndTermsIter) Reset(
-	ctx context.Context,
-	reader segment.Reader,
-	opts fieldsAndTermsIteratorOpts,
-) error {
-	*fti = fieldsAndTermsIterZeroed
-	fti.reader = reader
-	fti.opts = opts
-	if reader == nil {
-		return nil
+	iter := &fieldsAndTermsIter{
```

Comment:
`Reset` has been removed since we don't reuse the iterator between segments in a block. since we eagerly search all segments in a block, we need an iterator per segment.